calicoctl/calico_ctl/pool.py

```diff
@@ -151,8 +162,18 @@ def ip_pool_range_add(start_ip, end_ip, version, ipip, masquerade):
             sys.exit(1)
 
     cidrs = netaddr.iprange_to_cidrs(start_ip, end_ip)
+    new_pools = []
     for ip_net in cidrs:
-        new_pool = IPPool(ip_net.cidr, ipip=ipip, masquerade=masquerade)
+        try:
+            new_pools.append(IPPool(ip_net.cidr, ipip=ipip, masquerade=masquerade))
+        except CidrTooSmallError:
+            print "IPv%s range must start and end on a %s prefix network." \
```

Comment:
I think this error message needs a little work.  I'm wondering if we should make it more transparent what this command does - i.e. inform the user that we are breaking the range into CIDRs and storing as CIDRs.  If we don't then the comment about the offending pool would seem a little odd.

Not sure the commend about starting and ending on a /26 prefix network is true.  I think it is more that the start and end ip address must have at least the lower n bits unset.  I've no idea what the terminology would be for that - although you could talk about an individual pool being too small.